extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsStorageDruidModule.java

```diff
@@ -126,5 +129,7 @@ public void configure(Binder binder)
     Binders.taskLogsBinder(binder).addBinding("hdfs").to(HdfsTaskLogs.class);
     JsonConfigProvider.bind(binder, "druid.indexer.logs", HdfsTaskLogsConfig.class);
     binder.bind(HdfsTaskLogs.class).in(LazySingleton.class);
+    LifecycleModule.register(binder, HdfsStorageAuthentication.class);
```

Comment:
IIRC you still have to bind the class in the manage lifecycle scope. I always have to hit lifecycle management from a few angles to get it to take correctly.